modules/standard/Regex.chpl

```diff
@@ -833,27 +837,35 @@ record regex {
   pragma "no doc"
   proc _search_match(text: exprType, anchor: c_int, param has_captures, ref captures):regexMatch
   {
-    var ret:regexMatch;
-    on this.home {
-      var pos:byteIndex;
-      var endpos:byteIndex;
-
-      pos = 0;
-      endpos = pos + text.numBytes;
-
-      var matches:_ddata(qio_regex_string_piece_t);
-      param nmatches = if has_captures then 1 + captures.size else 1;
-      matches = _ddata_allocate(qio_regex_string_piece_t, nmatches);
-      var got:bool;
-      got = qio_regex_match(_regex, text.localize().c_str(), text.numBytes,
-                             pos:int, endpos:int, anchor,
-                             matches, nmatches);
-      // Now try to coerce the read strings into the captures.
-      if has_captures then _handle_captures(text, matches, nmatches, captures);
-      // Now return where we matched.
-      ret = new regexMatch(got, matches[0].offset:byteIndex, matches[0].len);
-      _ddata_free(matches, nmatches);
+    /* This pattern is a bit ugly, but we'd like to avoid
+    unnecessarily retaining this when local. and we need to maintain
+    the lifetime of the copied regex when remote.  The first
+    assignment to localRegex does a comm_get, even with split init
+    inside a if home == here. It would be nice if the compiler would
+    strength reduce that pattern.  */
+    var localRegex = _regex;
+    var regexCopy:regex(exprType);
+    if home != here {
+      regexCopy = this;
+      localRegex = regexCopy._regex;
     }
```

Comment:
Note that this adds an extra remote GET when the `home != here`. This isn't a case we care about optimizing a lot, but seems like we could avoid that with something like:

```chpl
    var regexCopy:regex(exprType);
    if home != here then regexCopy = this;
    const localRegex = if home != here then regexCopy._regex else _regex;
```

Reply:
Do you mean the GET for transferring the regex? I do see one less `chpl_gen_comm_get` with your version, but that is removing the `get` for the first line `var localRegex = _regex;`. So I am inclined to use your version, though I'm still figuring out how/why its different

Reply:
When `home != here`, previously,  `var localRegex = _regex` would do a GET, and then `localRegex` gets overridden by `localRegex = regexCopy._regex`, so it was a wasted GET.

In the pattern I suggested I don't think there's ever that extra GET. We may still be generating a call to `chpl_get_comm_get`, but I'm pretty sure there won't be a network remote GET at runtime, though it could be worth double checking the comm counts.